### DISCUSSION TITLE
How is it possible to render a element when its style or fill attribute is changed ?

### DISCUSSION THREAD

**Discussion by pierreandreline** [ElinamLLC/SharpVectors#183] [Q&A]
In a SvgPictureBox, I need to dynamically change the colors of svg elements based on realtime data. For example :
```csharp
myElement.SetAttribute("fill", "#00FF00"); 
// or 
myElement.SetAttribute("style", "fill: #00FF00");
```
But the change is not rendered in the SvgPictureBox instance.
Is there any other solution to change these attributes? or should I call a method to render after the change ? Thanks for any help

**Comment by paulushub** [comment]
@pierreandreline Please I need some further information, how are you using this  control? (have you extended it, how do you get the `myElement`, etc)

**Comment by pierreandreline** [comment]
I'm using SharpVectors lib v1.7.1
I created a small test application with a WinForm.
I added a SvgPictureBox instance in the WinForm.
```csharp
SvgPictureBox svg = new SvgPictureBox();
```

I add this instance to the WinForm and load a svg file with only on rectangle :
```csharp
this.Controls.Add(svg);
svg.Load(@"rect.svg");
```

Here's the rect.svg file content :
```xml
<svg xmlns="http://www.w3.org/2000/svg">
  <rect id="r" x="20" y="20" width="100" height="100" fill="#FF0000"/>
</svg>
```

The rectangle is displayed correctly in red.

On a button click, I try to change the fill color of then rectangle to green :
```csharp
SvgElement r = svg.Window.Document.GetElementById("r") as SvgElement;
r.SetAttribute("fill", "#00FF00");
```

The attribute change works but nothing changes on screen.
But when I read the attribute, I can check it is well changed :
```csharp
Console.WriteLine(r.GetAttribute("fill"));  // display "#00FF00" ok
```

**Reply by paulushub** [reply]
I have created an issue for this #184 

I can verified that it is not working, sorry. The reason is, the rendered image is cached, 
it is only refreshed if properties like the size changes.
I will post a fix, which will override the `Refresh` method to render it.

**Reply by pierreandreline** [reply]
ok, thank you very much for considering my request

**Reply by paulushub** [reply]
@pierreandreline  The issue is now fixed in the repository, but few more fixes before I release the Nuget package. You can verify the fix for this issue now, thank you.

**Reply by pierreandreline** [reply]
Thank you ! I have some problems to open projects in Visual Studio (2019) (*"C:\Program Files (x86)\dotnet\sdk\2.0.0\Sdks\Microsoft.NET.Sdk.WindowsDesktop\Sdk" not found*), so I'm not able yet to build the library. Dotnet 2.0.0 is installed but without WindowsDesktop folder (x86) in the sdks. I'll check that to fix it on my computer as soon as I can and I keep you informed. Thanks again.
